### ISSUE TITLE
Histogram bellcurve misaligned with standard deviation X-Axis with changes to chart element's width

### ISSUE THREAD

**Issue by erichiggins** [highcharts/highcharts#7604]
#### Expected behaviour
The distribution bell curve should maintain the same alignment with the X-Axis, regardless of element width.

#### Actual behaviour
The distribution bell curve is not always aligned with the mean/average on the X-Axis as the width of the containing element changes.

#### Live demo with steps to reproduce
https://jsfiddle.net/e0yz8q4o/
Load the JS Fiddle above and slide the width of the output container.
Note the alignment of the top of the bell curve with the top X-Axis (which shows difference from the standard deviation).

## This is pretty close to correct alignment.
<img width="636" alt="screen shot 2018-01-02 at 2 14 45 pm" src="https://user-images.githubusercontent.com/507145/34502229-76667724-efc7-11e7-95df-ff131581c5f6.png">

## Nope.
<img width="1009" alt="screen shot 2018-01-02 at 2 14 54 pm" src="https://user-images.githubusercontent.com/507145/34502238-7c132dc0-efc7-11e7-8269-558975cf516b.png">

## Wat?
<img width="462" alt="screen shot 2018-01-02 at 2 15 12 pm" src="https://user-images.githubusercontent.com/507145/34502244-8014ae3a-efc7-11e7-827d-ec32df279b26.png">

#### Affected browser(s)
Chrome (OSX)

**Comment by TorsteinHonsi** [comment]
The X axes have aligned ticks by default, which makes them go to great length to keep the same number of ticks. This is prevented by setting `alignTicks` to false on the X axis level: https://jsfiddle.net/highcharts/e0yz8q4o/1/

@morganfree 
* Should we make our samples have `alignTicks` false to prevent this?
* The last screenshot above must be a bug?

**Comment by erichiggins** [comment]
Thanks for the additional info, @TorsteinHonsi !
I took a look at the updated JSFiddle you provided with `alignTicks: false`, but it doesn't seem to actually resolve the issue.  In a normal distribution curve, I'd expect the middle of the bell curve to be centered on the mean (average), and to be aligned with zero on the axis which displays standard deviations from the mean. 

I believe this is demonstrated well in the [interval and points in interval example](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/plotoptions/bellcurve-intervals-pointsininterval), where `μ` is the mean and `σ` is the standard deviation

**Comment by morganfree** [comment]
@erichiggins 

In the demo both axes are hidden but, if they appear, the ticks from the top axis are not aligned as you would want. The bellcurve series is aware only about one axis and it is not concerned about computing/aligning other axes - use cases can differ so I believe that configuring other axes should be left to users.

In your case, if I understand it correctly, you can build third axis connected to the bellcurve axis with the ticks in specific positions and with the specific labels for the ticks.

    let labels = {}

    window.chart = Highcharts.chart('container', {
    chart: {
      events: {
      // it needs to be done on load event - the information from the computed series are needed
        load () { 
           const {
             data,
             mean,
             standardDeviation,
             options: { pointsInInterval }
           } = this.get('bellcurve')

          // make the ticks - each ticks represents the unit of the standard deviation
           const tickPositions = data
             .filter((point, i) => !(i % pointsInInterval))
             .map(point => point.x)
             
           // make the dictionary for the tick labels, it will be used in the labels formatter
           labels = tickPositions.reduce((labels, tick) => {
             const value = (Math.abs(tick - mean) / standardDeviation).toFixed()
             
             labels[tick] = value
             // labels[tick] =  !Number(value) ? 'μ' : `${value}σ`
             
             return labels
           }, {})
           
           // update tickPositions
           this.xAxis[2].update({ tickPositions })
        }
      }
    },
    xAxis: [{
        title: { text: 'Data' },
        opposite: true,
        alignTicks: false,
        visible: false
    }, {
        title: { text: 'Bell curve' },
        alignTicks: false
    }, {
     // define third axis
      linkedTo: 1,
      alignTicks: false,
      opposite: true,
      labels: {
        // make use of the labels dictionary
        formatter () {
          return labels[this.value] || this.value
        }
      }
    }],

live example: https://jsfiddle.net/dqfp48q6/

@TorsteinHonsi 

Yes, we can set `alignTicks` to false to prevent it (axes are hidden anyway) but It seems to be a regression bug, 6.0.3 -> 6.0.4

6.0.3: https://jsfiddle.net/30eyoxx8/
6.0.4: https://jsfiddle.net/30eyoxx8/1/

**Comment by TorsteinHonsi** [comment]
It fails since https://github.com/highcharts/highcharts/commit/4b0f2e30d.

> Yes, we can set alignTicks to false to prevent it (axes are hidden anyway) but It seems to be a regression bug, 6.0.3 -> 6.0.4

Do you agree that we set `alignTicks` to false in our demos, or is that not recommended? By setting it to false in the demos, we still leave it to the users, but we show that there's an option.

**Comment by morganfree** [comment]
I agree - we can set `alignTicks` to false.

**Comment by KacperMadej** [comment]
This looks like a more general issue caused by the mentioned commit - demo: https://jsfiddle.net/BlackLabel/9exu6941/1/

**Comment by KacperMadej** [comment]
After further testing turns out that for `alignTicks` to work correctly axes should have `startOnTick` and `endOnTick` set to `true` and `xAxis` by default has both set to false causing the problem more visible after [the commit](https://github.com/highcharts/highcharts/commit/4b0f2e30d).

The `startOnTick` and `endOnTick` set to `false` for `yAxis` will prevent axes to be aligned properly.
Demo for before the commit - version 6.0.3 (not aligned): http://jsfiddle.net/BlackLabel/w7mfjy7h/
Demo for after the commit - version 6.0.4 (series plot outside of plot area): http://jsfiddle.net/BlackLabel/w7mfjy7h/1/
6.0.4 with `startOnTick` and `endOnTick` set to `true` (all ok): http://jsfiddle.net/BlackLabel/w7mfjy7h/2/

**tl;dr**
@TorsteinHonsi 
This bug could be resolved by (1) adding info to the docs about `startOnTick` and `endOnTick` being required to be set to `true` and optionally by (2) forcing it at options level in case `alignTicks` is set to `true`. Also, (3) `alignTicks` is not documented on axis options level in API reference.

How to handle this?
- [ ] 1
- [ ] 2
- [ ] 3

Of course I'm happy to handle this any other way you choose.

**Comment by TorsteinHonsi** [comment]
What about another approach, simply disable `alignTicks` when `startOnTick` or `endOnTick` are false`? I think that would be _least_ surprising/confusing to the implementer. Plus it needs to be mentioned in the docs for `alignTicks`. As for implementation, it would be easy to check these options from within the tick extension logic.

What do you think?

**Comment by KacperMadej** [comment]
Looks good for me. I can handle this if you want.

What do you think @morganfree , @erichiggins ?

**Comment by erichiggins** [comment]
Sounds good to me. Thanks for the quick and thoughtful discussion on this issue!
